compiler/src/dotty/tools/dotc/transform/init/Errors.scala

```diff
@@ -89,20 +82,16 @@ object Errors {
   }
 
   /** Promote a value under initialization to fully-initialized */
-  case class UnsafePromotion(msg: String, source: Tree, trace: Seq[Tree], errors: Errors) extends Error {
-    assert(errors.nonEmpty)
+  case class UnsafePromotion(msg: String, source: Tree, trace: Seq[Tree], error: Error) extends Error {
 
     override def issue(using Context): Unit =
       report.warning(show, source.srcPos)
 
     def show(using Context): String = {
       var index = 0
       "Promoting the value to fully-initialized is unsafe. " + msg + "\n" + stacktrace +
```

Comment:
Let's change the sentence to: `Cannot prove that the value is fully-initialized`.